core/controllers/suggestion_test.py

```diff
@@ -475,3 +702,218 @@ def test_accept_question_suggestion(self):
+                    .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+                'question_dict': self.question_dict,
+                'skill_id': None
+            }, 'score_category', thread_id)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', False):
+            self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX, self.topic_id,
+                    thread_id),
+                {}, csrf_token=csrf_token, expected_status_int=404)
+
+        self.logout()
+
+    def test_suggestion_to_topic_handler_with_invalid_target_type(self):
+        self.login(self.OWNER_EMAIL)
+
+        thread_id = feedback_services.create_thread(
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            self.author_id, 'description', '', has_suggestion=True)
+
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_ADD_QUESTION,
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            1, suggestion_models.STATUS_IN_REVIEW, self.author_id,
+            None, {
+                'cmd': (
+                    question_domain
+                    .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+                'question_dict': self.question_dict,
+                'skill_id': None
+            }, 'score_category', thread_id)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX,
+                    self.topic_id, thread_id),
+                {}, csrf_token=csrf_token, expected_status_int=400)
+
+        self.assertEqual(
+            response['error'],
+            'This handler allows actions only on suggestions to topics.')
+
+        self.logout()
+
+    def test_suggestion_to_topic_handler_with_invalid_target_id(self):
+        self.login(self.OWNER_EMAIL)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        self.save_new_topic(
+            'topic_id', self.owner_id, 'Name1', 'Description',
+            [], [], [], [], 1)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json(
```

Comment:
Doesn't this request need an "action" parameter passed in?

Reply:
If we change the order of the checks in the controller later, this test would fail while it shouldn't.

Reply:
Done